### PR TITLE
User can see a list of their requests

### PR DESCRIPTION
**Pull request by kermit-klein**
[PT LINK](https://www.pivotaltracker.com/story/show/173334319)

- Add index action to my_requests
- Add serializer for index action
- Add routes
- Add happy and sad path tests

### CODE REVIEW

**Review by PaulineBA — COMMENTED** [review]
Good one ;-)

File: app/controllers/api/my_request/requests_controller.rb [hunk 1]
```diff
@@ -1,10 +1,19 @@
...
 class Api::MyRequest::RequestsController < ApplicationController
-  before_action :authenticate_user!, only: %i[create update]
+  before_action :authenticate_user!, only: %i[create update index]
   before_action :karma?, only: [:create]
   rescue_from ArgumentError, with: :render_error_message
 
+  def index
+    requests = Request.where(requester: current_user).order('id DESC')
+    if requests == []
+      render json: { message: 'There are no requests to show' }, status: 404
```
Comment (erikbjoern) [inline app/controllers/api/my_request/requests_controller.rb:11]: I think this should be 204. It's good that you have a message for it anyway!

File: spec/requests/api/my_request/requests/index_spec.rb [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+        it ':created_at' do
+          expect(response_json['requests'][0]).not_to have_key 'created_at'
+        end
+        it ':requester' do
+          expect(response_json['requests'][0]).not_to have_key 'requester'
+        end
+        it ':description' do
+          expect(response_json['requests'][0]).not_to have_key 'description'
+        end
+      end
```
Comment (erikbjoern) [inline spec/requests/api/my_request/requests/index_spec.rb:51]: Aren't we going to use Thomas's serializer spec here instead?

File: spec/requests/api/my_request/requests/index_spec.rb [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    before do
+      get '/api/my_request/requests', headers: headers
+    end
+
+    describe 'successfully gets the requests' do
+      it 'has a 200 response' do
+        expect(response).to have_http_status 200
+      end
+
+      it 'contains all the requests' do
```
Comment (erikbjoern) [inline spec/requests/api/my_request/requests/index_spec.rb:23]: We should have a test to assert that we only get our own requests, not all

File: spec/requests/api/my_request/requests/index_spec.rb [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+    before do
+      get '/api/my_request/requests', headers: headers_2
+    end
+
+    it 'has 404 response' do
+      expect(response).to have_http_status 404
+    end
+
+    it 'responds error message' do
+      expect(response_json['message']).to eq 'There are no requests to show'
```
Comment (erikbjoern) [inline spec/requests/api/my_request/requests/index_spec.rb:79]: Okay, I guess this works as that assertion I commented about above

**Review by erikbjoern — CHANGES_REQUESTED** [review]

**Review by erikbjoern — APPROVED** [review]

**Review by ViaMarcus — APPROVED** [review]

**Pull request merged**